plugins/cisco_umbrella_enforcement/komand_cisco_umbrella_enforcement/connection/connection.py

```diff
@@ -1,35 +1,27 @@
-import komand
-from .schema import ConnectionSchema
+import insightconnect_plugin_runtime
+
+from .schema import ConnectionSchema, Input
+from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
 
 # Custom imports below
-from ..api import *
+from ..util.api import CiscoUmbrellaEnforcementAPI
 
 
-class Connection(komand.Connection):
+class Connection(insightconnect_plugin_runtime.Connection):
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
+        self.client = None
 
     def connect(self, params={}):
         self.logger.info("Connect: Connecting..")
 
-        key = params.get("api_key").get("secretKey")
-        if not key:
-            self.logger.error("Connection: connect: Empty key")
-            raise Exception("Connection: connect: Empty key")
-
-        key = key.lower()
-        p = re.compile("[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}")
-        m = p.match(key)
-        if not m:
-            self.logger.error("Connection: connect: Wrong key")
-            raise Exception("Connection: connect: Wrong key")
+        key = params.get(Input.API_KEY).get("secretKey")
 
-        self.api = Cisco_Api(key)
-        self.testConnection()
+        self.client = CiscoUmbrellaEnforcementAPI(key)
 
-    def testConnection(self):
+    def test(self):
         try:
-            self.api.get_domains()
-        except Exception:
-            self.logger.error("Connection: connect: Wrong key")
-            raise Exception("Connection: connect: Wrong key")
+            self.client.get_domains()
+            return {"success": True}
+        except PluginException as error:
+            raise ConnectionTestException(cause=error.cause, assistance=error.assistance, data=error.data)
```

Comment:
did you not say earlier that we dont keep test functions in these kinda files anymore?

Reply:
We dont for actions we do for connections